### PR TITLE
Made sentence selection reader add random negative sentences

### PR DESCRIPTION
**Pull request by matt-gardner**

### CODE REVIEW

File: deep_qa/data/dataset_readers/squad_sentence_selection_reader.py [hunk 1]
```diff
@@ -20,26 +22,107 @@ def main():
...
+    ----------
+    output_directory: str, optional (default=None)
+        If you want the output stored somewhere other than in a ``processed/`` subdirectory next to
+        the input, you can override the default with this parameter.
+    negative_sentence_selection: str, optional (default="paragraph")
+        A comma-separated list of methods to use to generate negative sentences in the data.
+
+        There are three options here:
+
+        (1) "paragraph", which means to use as negative sentences all other sentences in the same
```
Comment (matt-gardner) [inline deep_qa/data/dataset_readers/squad_sentence_selection_reader.py]: I'm about to push another commit with more options that will make the motivation for this a lot more obvious.  And in general, more options to play around with stuff doesn't hurt anything, so why assume that "paragraph" is always done, if it's not already written that way?

**Comment by nelson-liu** [comment]
looks like travis is unhappy

**Comment by matt-gardner** [comment]
Ugh...  Seeing if setting PYTHONHASHSEED in the travis test script helps...

**Comment by matt-gardner** [comment]
Looks like that worked.

File: deep_qa/data/dataset_readers/squad_sentence_selection_reader.py (hunk 1 above)
Comment (nelson-liu) [inline deep_qa/data/dataset_readers/squad_sentence_selection_reader.py]: > And in general, more options to play around with stuff doesn't hurt anything, so why assume that "paragraph" is always done, if it's not already written that way?

sure, I personally feel like more options make things harder to break / use :) but I understand the rationale behind these options and I'm fine with it.

File: tests/data/dataset_readers/squad_sentence_selection_reader_test.py [hunk 2]
```diff
@@ -11,18 +14,32 @@ class TestSquadSentenceSelectionReader(DeepQaTestCase):
...
-                   "Immediately in front of the Main Building and facing it, is a copper "
-                   "statue of Christ with arms upraised with the legend \\\"Venite Ad Me Omnes\\\". "
-                   "Next to the Main Building is the Basilica of the Sacred Heart. Immediately "
-                   "behind the basilica is the Grotto, a Marian place of prayer and reflection. "
-                   "It is a replica of the grotto at Lourdes, France where the Virgin Mary "
-                   "reputedly appeared to Saint Bernadette Soubirous in 1858. At the end of the "
-                   "main drive (and in a direct line that connects through 3 statues and the Gold "
-                   "Dome), is a simple, modern stone statue of Mary.")
+        # pylint: disable=bad-continuation
+        self.sentences = [
```
Comment (nelson-liu) [inline tests/data/dataset_readers/squad_sentence_selection_reader_test.py]: thanks, this is a lot cleaner

**Pull request merged**